grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1070,6 +1070,16 @@ class StockMovementService {
             log.info "Available items: ${availableItems}"
             List<SuggestedItem> suggestedItems = getSuggestedItems(availableItems, quantityRequired)
             log.info "Suggested items " + suggestedItems
+
+            def shipmentItems = ShipmentItem.findAllByRequisitionItem(requisitionItem)
```

Comment:
it should not be done here, it may cause some issues and create the shipment items two times
move this to `void createPicklist(String id)` method (in line 1026), so it's done only on picklist refresh,
also you can use `removeShipmentItemsForModifiedRequisitionItem` method for this

Reply:
Corrected